recipes/cassandra-cpp-driver/all/patches/2.15.3/fix-cmake.patch

```diff
@@ -0,0 +1,352 @@
+ CMakeLists.txt                               |  30 +----
+ src/CMakeLists.txt                           | 190 ++++++++++++---------------
+ src/third_party/curl/CMakeLists.txt          |  16 +--
+ src/third_party/hdr_histogram/CMakeLists.txt |  15 +--
+ 4 files changed, 97 insertions(+), 154 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c6150948b..6d9f13965 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,6 +1,3 @@
+-cmake_minimum_required(VERSION 2.8.12)
+-project(cassandra C CXX)
+-
+ set(CASS_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+ set(CASS_SRC_DIR "${CASS_ROOT_DIR}/src")
+ set(CASS_INCLUDE_DIR "${CASS_ROOT_DIR}/include")
+@@ -29,8 +26,6 @@ endif()
+ 
+ option(CASS_BUILD_EXAMPLES "Build examples" OFF)
+ option(CASS_BUILD_INTEGRATION_TESTS "Build integration tests" OFF)
+-option(CASS_BUILD_SHARED "Build shared library" ON)
+-option(CASS_BUILD_STATIC "Build static library" OFF)
+ option(CASS_BUILD_TESTS "Build tests" OFF)
+ option(CASS_BUILD_UNIT_TESTS "Build unit tests" OFF)
+ option(CASS_DEBUG_CUSTOM_ALLOC "Debug custom allocator" OFF)
+@@ -42,7 +37,6 @@ option(CASS_USE_BOOST_ATOMIC "Use Boost atomics library" OFF)
+ option(CASS_USE_KERBEROS "Use Kerberos" OFF)
+ option(CASS_USE_LIBSSH2 "Use libssh2 for integration tests" OFF)
+ option(CASS_USE_OPENSSL "Use OpenSSL" ON)
+-option(CASS_USE_STATIC_LIBS "Link static libraries when building executables" OFF)
+ option(CASS_USE_STD_ATOMIC "Use C++11 atomics library" OFF)
+ option(CASS_USE_ZLIB "Use zlib" ON)
+ option(CASS_USE_TIMERFD "Use timerfd (Linux only)" ON)
+@@ -60,28 +54,12 @@ if(CASS_BUILD_INTEGRATION_TESTS OR CASS_BUILD_UNIT_TESTS)
+ endif()
+ 
+ # Determine which driver target should be used as a dependency
+-set(PROJECT_LIB_NAME_TARGET cassandra)
+-if(CASS_USE_STATIC_LIBS OR
+-   (WIN32 AND (CASS_BUILD_INTEGRATION_TESTS OR CASS_BUILD_UNIT_TESTS)))
+-  set(CASS_USE_STATIC_LIBS ON) # Not all driver internals are exported for test executable (e.g. CASS_EXPORT)
+-  set(CASS_BUILD_STATIC ON)
+-  set(PROJECT_LIB_NAME_TARGET cassandra_static)
+-endif()
+-
+-# Ensure the driver is configured to build
+-if(NOT CASS_BUILD_SHARED AND NOT CASS_BUILD_STATIC)
+-  message(FATAL_ERROR "Driver is not Configured to Build: Ensure shared and/or static library is enabled")
+-endif()
+-
+-if(CASS_DEBUG_CUSTOM_ALLOC AND CASS_USE_STATIC_LIBS)
+-  message(WARNING "Debugging the custom allocator while static linking the library can cause your application to fail")
+-endif()
++set(PROJECT_LIB_NAME_TARGET ${PROJECT_NAME})
+ 
+ #------------------------
+ # Dependencies
+ #------------------------
+ 
+-include(Dependencies)
+ include(ClangFormat)
+ 
+ #------------------------
+@@ -229,3 +207,9 @@ endif()
+ if(CASS_BUILD_INTEGRATION_TESTS OR CASS_BUILD_UNIT_TESTS)
+   add_subdirectory(tests)
+ endif()
++
++#------------------------
++# Install licenses
++#------------------------
++
++install(FILES LICENSE.txt DESTINATION licenses)
```

Comment:
You can simply use `self.copy` in conanfile.py, instead of patching the cmake script.

This make the patch harder to port to next cassandra-cpp-driver releases.

Reply:
[Fixed](https://github.com/conan-io/conan-center-index/pull/3110/commits/eaa640430ae6008d0df8a0b86d894f17ca4e98da)